src/rgw/rgw_common.cc

```diff
@@ -412,10 +410,9 @@ void req_info::init_meta_info(bool *found_bad_meta)
         }
         name_low[j] = 0;
 
-        map<string, string>::iterator iter;
-        iter = x_meta_map.find(name_low);
-        if (iter != x_meta_map.end()) {
-          string old = iter->second;
+        auto it = x_meta_map.find(name_low);
+        if (it != x_meta_map.end()) {
+          string old = it->second;
           int pos = old.find_last_not_of(" \t"); /* get rid of any whitespaces after the value */
           old = old.substr(0, pos + 1);
```

Comment:
I vote for `trim_right`. We're already using `boost::algorithm::trim` in couple of places.